grub-core/loader/multiboot_elfxx.c

```diff
@@ -106,12 +115,54 @@ CONCAT(grub_multiboot_load_elf, XX) (mbi_load_data_t *mld)
 		    (long) mld->align, mld->preference, load_size,
 		    mld->avoid_efi_boot_services);
 
-      if (load_size > mld->max_addr || mld->min_addr > mld->max_addr - load_size)
+      if (grub_slaunch_platform_type () == SLP_INTEL_TXT)
+	  {
+#ifndef GRUB_USE_MULTIBOOT2
+	    return grub_error (GRUB_ERR_BAD_OS, "Only multiboot2 supported for slaunch");
+#endif
+	    /*
+	     * We allocate the the binary together with page tables and MBI
+	     * to make one contiguous block for MLE. We have to align up to
+	     * PMR (2MB).
+	     */
+	    total_size = ALIGN_UP(load_size, MULTIBOOT_TAG_ALIGN); 
+	    total_size += GRUB_MULTIBOOT (get_mbi_size)();
+	    total_size = ALIGN_UP(total_size, GRUB_TXT_PMR_ALIGN);
+
+	    slparams->mle_size = total_size;
+
+	    slparams->mle_ptab_size = grub_txt_get_mle_ptab_size (total_size);
+	    slparams->mle_ptab_size = ALIGN_UP (slparams->mle_ptab_size, GRUB_TXT_PMR_ALIGN);
+
+	    total_size += slparams->mle_ptab_size;
+	    /* Do not go below GRUB_TXT_PMR_ALIGN. */
+	    if (mld.align > GRUB_TXT_PMR_ALIGN)
```

Comment:
```suggestion
	    if (mld->align > GRUB_TXT_PMR_ALIGN)
```
Here and every other use of `mld`

Reply:
Fixed